sycl/source/detail/memory_manager.cpp

```diff
@@ -255,34 +289,44 @@ void copyH2D(SYCLMemObjI *SYCLMemObj, char *SrcMem, QueueImplPtr,
   assert(SYCLMemObj && "The SYCLMemObj is nullptr");
 
   const RT::PiQueue Queue = TgtQueue->getHandleRef();
-  // Adjust first dimension of copy range and offset as OpenCL expects size in
-  // bytes.
-  DstSize[0] *= DstElemSize;
   const detail::plugin &Plugin = TgtQueue->getPlugin();
-  if (SYCLMemObj->getType() == detail::SYCLMemObjI::MemObjType::BUFFER) {
-    DstOffset[0] *= DstElemSize;
-    SrcOffset[0] *= SrcElemSize;
-    SrcAccessRange[0] *= SrcElemSize;
-    DstAccessRange[0] *= DstElemSize;
-    SrcSize[0] *= SrcElemSize;
 
+  detail::SYCLMemObjI::MemObjType MemType = SYCLMemObj->getType();
+  term_positions SrcPos, DstPos;
+  prepTermPositions(SrcPos, DimSrc, MemType);
+  prepTermPositions(DstPos, DimDst, MemType);
+
+  //  If the Dimension is 1, and being called by ~SYCLMemObjT,
+  //  thenDstAccessRange[0] and DstSize[0] will already sized to bytes with
```

Comment:
Could you please elaborate more on this? Is this comment still relevant?

Reply:
The comment is relevant, but maybe I'm the only one who cares?   If you were debugging the memory handling of a buffer, then you might be surprised to see different values appearing for the same buffer, and the reason is that we schedule a slightly different mem operation when destructing.

Reply:
I wrote a book on the topic and replaced this comment with an amazon affiliate link. win win.

Reply:
The comments stay forever, the affiliate links do not... :-)